src/wp-includes/class-wp-theme-json.php

```diff
@@ -2128,9 +2154,76 @@ protected static function compute_style_properties( $styles, $settings = array()
 			}
 		}
 
+		if ( $can_use_cached ) {
+			self::set_compute_style_properties_cache( $cache_key, $declarations );
+		}
+
 		return $declarations;
 	}
 
+	/**
+	 * Sets the cache for the compute style properties.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param string $cache_key The cache key.
+	 * @param array  $declarations The declarations to cache.
+	 *
+	 * @return array The cached declarations.
+	 */
+	private static function set_compute_style_properties_cache( $cache_key, $declarations ) {
+		self::$cached_compute_style_properties[ $cache_key ] = $declarations;
+		set_site_transient( 'wp_compute_style_properties', self::$cached_compute_style_properties, HOUR_IN_SECONDS );
+		self::$cached_compute_style_properties;
+	}
+
+	/**
+	 * Gets the cache for the compute style properties.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return array The cached declarations.
+	 */
+	private static function get_compute_style_properties_cache() {
+		self::$cached_compute_style_properties = get_site_transient( 'wp_compute_style_properties' );
+
+		if ( ! is_array( self::$cached_compute_style_properties ) ) {
+			self::$cached_compute_style_properties = array();
+		}
+
+		return self::$cached_compute_style_properties;
+	}
+
+	/**
+	 * Clears the cache for the compute style properties.
+	 *
+	 * @since 6.6.0
+	 */
+	public static function clear_compute_style_properties_cache() {
+		self::$cached_compute_style_properties = null;
+		delete_site_transient( 'wp_compute_style_properties' );
+	}
+
+	/**
+	 * Gets the hash of the global settings.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return string The hash of the global settings.
+	 */
+	private static function wp_get_global_settings_hash() {
+		$cache_group = 'theme_json';
+		$cache_key   = 'wp_get_global_settings_hash';
+		$cache       = wp_cache_get( $cache_key, $cache_group );
+		if ( $cache ) {
+			return $cache;
+		}
+
+		$hash = md5( wp_json_encode( wp_get_global_settings() ) );
```

Comment:
This would create the same circular dependency issue that was raised in [this comment](https://github.com/WordPress/wordpress-develop/pull/5567#discussion_r1452068125). If the computed style properties we want to cache are related to the final computed `WP_Theme_JSON` object, then we would need to move this cache to the place where that final object is accessed. It looks like this happens in `wp_add_global_styles_for_blocks()` where the computed `WP_Theme_JSON` is returned by `WP_Theme_JSON_Resolver::get_merged_data()` and then that instance is used to get a list of block nodes and loop through all of the blocks to get the styles for each block.

Otherwise, we need to look into what theme json property from `wp_get_global_settings()` could be used as part of the cache strategy here.